querqy-core/src/main/java/querqy/rewrite/contrib/ReplaceRewriterParser.java

```diff
@@ -0,0 +1,102 @@
+
+    public ReplaceRewriterParser(final InputStreamReader inputStreamReader,
+                                 final boolean ignoreCase,
+                                 final String inputDelimiter,
+                                 final QuerqyParser querqyParser) {
+
+        this.inputStreamReader = inputStreamReader;
+        this.ignoreCase = ignoreCase;
+        this.inputDelimiter = inputDelimiter;
+        this.querqyParser = querqyParser;
+    }
+
+    public TrieMap<List<ComparableCharSequence>> parseConfig() throws IOException {
+
+        TrieMap<List<ComparableCharSequence>> trieMap = new TrieMap<>();
+
+        try (BufferedReader bufferedReader = new BufferedReader(inputStreamReader)) {
+
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+
+                line = line.trim();
+                if (line.isEmpty() || line.startsWith("#")) {
+                    continue;
+                }
+
+                final String[] lineSplit = line.split(OPERATOR);
+                if (lineSplit.length != 2) {
+                    throw new RuleParseException(ERROR_MESSAGE);
+                }
+
+                final String fullInput = lineSplit[0].trim();
+                final String output = lineSplit[1].trim();
+                final List<String> inputs = Arrays.stream(fullInput.split(this.inputDelimiter))
+                        .map(String::trim)
+                        .filter(term -> !term.isEmpty())
+                        .collect(Collectors.toList());
+
+                if (output.isEmpty() || inputs.isEmpty()) {
+                    throw new RuleParseException(ERROR_MESSAGE);
+                }
+
+                final List<ComparableCharSequence> outputList = this.querqyParser.parse(output).getClauses().stream()
+                        .map(booleanClause -> (DisjunctionMaxQuery) booleanClause)
+                        .flatMap(disjunctionMaxQuery -> disjunctionMaxQuery.getTerms().stream())
+                        .map(Term::getValue)
+                        .collect(Collectors.toCollection(LinkedList::new));
+
+                final List<ComparableCharSequence> inputList = inputs.stream()
+                        .map(this.querqyParser::parse)
+                        .map(query -> query.getClauses().stream()
+                                .map(booleanClause -> (DisjunctionMaxQuery) booleanClause)
+                                .map(DisjunctionMaxQuery::getTerms)
+                                .flatMap(Collection::stream)
+                                .map(Term::getValue)
+                                .map(seq -> ignoreCase ? new LowerCaseCharSequence(seq) : seq)
+                                .collect(Collectors.toList()))
+                        .map(sequenceList -> new CompoundCharSequence(TOKEN_SEPARATOR, sequenceList))
+                        .collect(Collectors.toList());
+
+                inputList.forEach(seq -> trieMap.put(seq, outputList));
```

Comment:
I think we want to forbid a configuration like this, i.e. two different replacements for the same input:
```
a => b
a => c
```
As far as I can see the configuration is not yet validated against this.